python/across_api/base/api.py

```diff
@@ -19,3 +23,167 @@
     },
     root_path="/labs/api/v1",
 )
+
+# FastAPI Depends definitions
+
+
+# Depends functions for FastAPI calls.
+async def optional_daterange(
+    begin: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="Begin",
+        ),
+    ] = None,
+    end: Annotated[
+        Optional[datetime],
+        Query(
+            description="Start time of period to be calculated.",
+            title="End",
+        ),
+    ] = None,
+) -> dict:
+    """
+    Helper function to convert begin and end to datetime objects.
+    """
+    if begin is None or end is None:
+        return {"begin": None, "end": None}
+    return {"begin": Time(begin), "end": Time(end)}
+
+
+OptionalDateRangeDep = Annotated[dict, Depends(optional_daterange)]
+
+
+# Depends functions for FastAPI calls.
+async def optional_length(
```

Comment:
Rename to duration.

Reply:
Fixed in [6ef39aa](https://github.com/nasa-gcn/gcn.nasa.gov/pull/1975/commits/6ef39aa41dc8bc60b462133ef6dbc16d2ee10f42)